### DISCUSSION TITLE
Wiping the margins

### DISCUSSION THREAD

**Discussion by Cicorione** [speedata/publisher#639] [Q&A]
Hi @pgundlach  this is the _grand-dessert_ that I left for last (hopefully)... 😁

This is something that I've always wanted to ask you but never dare... 🙈🙊

Here is the thing: I need to draw these gray lines with the products (picture below) but as you can notice these lines lie within the margins, and here is the problem.

![agl-layout-concept](https://github.com/user-attachments/assets/517de0e7-c081-4646-a450-68f7eecdf1e9)

I haven't figured hop to place these lines yet, but this template is divided into a grid and into each cell there is a group. So, for each group I might decide, for instance, to draw a line on the right side or the bottom side, whether if the layout in use is vertical or horizontally. 

Either way, at a certain point, these lines will go over the page margins and this is, of course, undesired. I might create, whether I am able within the group to calculate the full page  width and height, a switch case to draw these lines only within the margins area; or I might put a frame that has an area that just fits within the margins, but this depends if within the group I have access to full page size; anyway both approaches looks very tedious.

If I have been able to explain properly the issue, I wonder if you ever taken in consideration to create a tag, or an attribute, that when is invoked to an object it prevents Publisher to draw it outside the margin.

Let say something that it would call `WipesMargin` and therefore:

`tag`
```xml
<WipesMargin>
  <PlaceObject row="X" column="Y">
    [...]
  </PlaceObject>
</WipesMargin>
```

or

`attribute`
```xml
<PlaceObject row="X" column="Y" wipesmargin="yes">
  [...]
</PlaceObject>
```

I do believe that is a powerful feature that will save to many a lot of headaches, especially when it is time to draw patterns.

If something similar already exist because you're already thoughts about that even better! Please let me know!

Meanwhile many thanks! 🙏

**Comment by pgundlach** [comment]
Good point, I will think about it.

**Comment by pgundlach** [comment]
I have create and issue #640 and will close this.
